tests/tests_pytorch/strategies/test_deepspeed.py

```diff
@@ -1041,46 +1038,6 @@ def training_step(self, batch, batch_idx):
         trainer.fit(model)
 
 
-@RunIf(min_cuda_gpus=1, standalone=True, deepspeed=True)
-def test_deepspeed_setup_train_dataloader(tmpdir):
-    """Test DeepSpeed works when setup is required to call in the DataModule."""
-
-    class TestSetupIsCalledDataModule(LightningDataModule):
-        def __init__(self):
-            super().__init__()
-            self._setup = False
-
-        def setup(self, stage: str) -> None:
-            self._setup = True
-
-        def train_dataloader(self):
-            assert self._setup
```

Comment:
This assertion failed during inferring the batch size, but our exception handling would catch it and emit the warning we assert below. Since we delayed the config initialization, this is no longer a limitation, as setup() gets called before we do that. 

See my comment above where I removed the exception handling in the strategy.